// Lab161-10c
// Try Catch
// You have been provided with a tryCatch1.java file. 
// The java program contained in the file is supposed to: 
// Print to the screen the sum of the contents of an array. 
// Print to the screen the product of the contents of an array. 
// Print to the screen the contents of the array which are supposed to be filled with user input.

import java.util.Scanner;
public class tryCatch1{
public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int[] array=new int [5]; //new integer array of size 5
        int sum=0; //will hold the value of the sum of the inputs
        int product=1; //will hold the value of the product of the inputs
        
        
        try{
        for(int i=0;i<array.length;i++){
            array[i]=sc.nextInt();
            }
        
        sc.close(); 
    
        for(int i=0;i<=array.length;i++){
            System.out.print(array[i] + " ");
            product=product*array[i];
            sum=sum+array[i];
            }
        }
        
        catch(ArrayIndexOutOfBoundsException e) {
            System.out.println();
            System.out.println("An error has occurred.");
        }
        
        finally{
        System.out.println("Product: "+product);
        System.out.println("Sum: "+sum);
        }
        
    }
}
